service/queue_test.go

```diff
@@ -1501,6 +1501,227 @@ func testQueue(t *testing.T, setup NewStorageFunc, tearDown func()) {
 			assert.Equal(t, int32(0), p.Total)
 	})
 
+	t.Run("QueueReload", func(t *testing.T) {
+		t.Run("BasicReload", func(t *testing.T) {
+			var queueName = random.String("queue-", 10)
+			d, c, ctx := newDaemon(t, 10*clock.Second, svc.Config{StorageConfig: setup()})
+			defer func() {
+				d.Shutdown(t)
+				tearDown()
+			}()
+
+			createQueueAndWait(t, ctx, c, &pb.QueueInfo{
+				RequestedPartitions: 1,
+				LeaseTimeout:        LeaseTimeout,
+				ExpireTimeout:       ExpireTimeout,
+				QueueName:           queueName,
+			})
+
+			// Call QueueReload on empty queue
```

Comment:
We MUST introduce a new `pb.QueueReloadRequest` and `pb.QueueReloadResponse`. We should NOT use `QueueClearRequest`. This allows us to make changes to the request body without needing to increment a new version of the QueueReload in the future.